stix/idl/io/stx_map2fits.pro

```diff
@@ -0,0 +1,285 @@
+
+  ; ******************** ADAPTED FOR STIX ************************
+
+  ; Check if the L1 file is given
+  if not keyword_set(path_sci_file) then begin
+    print, ''
+    print, ' >>>>>>>>>> path_sci_file has to be defined!! <<<<<<<<<<'
+    print, ''
+    message, 'Please, define path_sci_file (path to the science file used for imaging)'
+  endif
+    
+  ; Check if the algorithm used is CLEAN.
+  ; If this is the case, then store only the CLEAN map and discard all other maps
+  this_id = in_map[0].id
+  if this_id.Contains('CLEAN') eq 1 and not keyword_set(all_clean) then begin
+    map = in_map[0]
+  endif else begin
+    map = in_map
+  endelse
+
+  ; *************************************************************
+
+   if ~valid_map(map) || is_blank(file) then begin
+    pr_syntax,'map2fits,map,file'
+    return
+   endif
+   np=n_elements(map)
+
+;-- form output file name
+
+   cd,curr=cdir
+   break_file,file,dsk,dir,dfile,dext
+   if trim(dsk+dir) eq '' then fdir=cdir else fdir=dsk+dir
+   if ~file_test(fdir,/write) then begin
+    err='No write access to - '+fdir
+    message,err,/info
+    return
+   endif
+   if trim(dext) eq '' then fext='.fits' else fext=dext
+   filename=concat_dir(fdir,dfile)+fext
+   filename=filename.Replace("'","")
+   
+   if verbose then message,'Writing map to - '+filename,/info
+   use_rtime=tag_exist(map,'rtime')
+   in_tags=tag_names(map)
+   def_map,rmap
+   def_fits=['SIMPLE','BITPIX','NAXIS','NAXIS1','NAXIS2','DATE','FILENAME','END',$
+             'COMMENT','HISTORY']
+   ignore=[tag_names(rmap),def_fits]
+   delvarx,header
+   for i=0,np-1 do begin
+
+;-- unpack data
+
+    unpack_map,map[i],data,xp,yp,dx=cdelt1,dy=cdelt2,xc=xcen,yc=ycen,$
+      nx=naxis1,ny=naxis2
+
+;-- scale data?
+
+    if keyword_set(byte_scale) then bscale,data,top=255
+
+;-- add header for the output array.
+
+    fxhmake,header,data,/date
+
+;-- add FITS parameters CRPIX, CRVAL, etc.
+
+    crpix1=comp_fits_crpix(xcen,cdelt1,naxis1,0)
+    crpix2=comp_fits_crpix(ycen,cdelt2,naxis2,0)
+
+    fxaddpar, header, 'ctype1', 'solar_x','Solar X (cartesian west) axis'
+    fxaddpar, header, 'ctype2', 'solar_y','Solar Y (cartesian north) axis'
+
+;    fxaddpar, header, 'cunit1', 'arcsecs','Arcseconds from center of Sun'
+;    fxaddpar, header, 'cunit2', 'arcsecs','Arcseconds from center of Sun'
+
+    fxaddpar, header, 'crpix1', crpix1, 'Reference pixel along X dimension'
+    fxaddpar, header, 'crpix2', crpix2, 'Reference pixel along Y dimension'
+
+    fxaddpar, header, 'crval1',0, 'Reference position along X dimension'
+    fxaddpar, header, 'crval2',0, 'Reference position along Y dimension'
+
+    fxaddpar, header, 'cdelt1',cdelt1,'Increments along X dimension'
+    fxaddpar, header, 'cdelt2',cdelt2,'Increments along Y dimension'
+
+    if use_rtime then obs_time=map[i].rtime else obs_time=map[i].time
+
+    fxaddpar,header,'date_obs',obs_time,'Observation date'
+    fxaddpar,header,'date-obs',obs_time,'Observation date'
```

Comment:
only one of these should be in here which ever on is in the SOLO meta data standard

Reply:
Thanks! I just looked into the EUI FITS file (L2) as an example and 'DATE-OBS' is used. I can use that then.